zgw/object-management/src/main/kotlin/com/ritense/objectmanagement/ObjectManagementPlugin.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.objectmanagement
+
+import com.ritense.objectenapi.client.ObjectRequest
+import com.ritense.objectenapi.client.ObjectWrapper
+import com.ritense.plugin.annotation.Plugin
+import com.ritense.plugin.annotation.PluginAction
+import com.ritense.plugin.annotation.PluginActionProperty
+import com.ritense.plugin.domain.ActivityType
+import java.net.URI
+import org.springframework.http.HttpStatus
+
+@Plugin(
+    key = "objectmanagement",
+    title = "Object Management",
+    description = "Provides object-management plugin actions for process-links"
+)
+class ObjectManagementPlugin {
+
+    @PluginAction(
+        key = "create-object",
+        title = "Create object",
+        description = "Create an object by Object Management",
+        activityTypes = [ActivityType.SERVICE_TASK_START]
+    )
+    fun createObject(objectRequest: ObjectRequest): ObjectWrapper {
+        TODO("Not implemented!")
+    }
+
+    @PluginAction(
+        key = "get-object",
+        title = "Get object",
+        description = "Get an object by Object Management",
+        activityTypes = [ActivityType.SERVICE_TASK_START]
+    )
+    fun getObject(objectUrl: URI): ObjectWrapper {
+        TODO("Not implemented!")
+    }
+
+    @PluginAction(
+        key = "update-object",
+        title = "Update object",
+        description = "Update an object by Object Management",
+        activityTypes = [ActivityType.SERVICE_TASK_START]
+    )
+    fun updateObject(objectUrl: URI, objectRequest: ObjectRequest): ObjectWrapper {
+        TODO("Not implemented!")
+    }
+
+    @PluginAction(
+        key = "delete-object",
+        title = "Delete object",
+        description = "Delete an object by Object Management",
+        activityTypes = [ActivityType.SERVICE_TASK_START]
+    )
+    fun deleteObject(@PluginActionProperty objectUrl: URI): HttpStatus {
```

Comment:
@anna-ritense @ThomasMinkeRitense 
This `delete-object` plugin action already exists in the ObjectenApiPlugin.

Also, if the other 3 plugin actions here would be implemented, they would only need use the ObjectenApiClient. Which means all plugin actions here, could be moved to the ObjectenApiPlugin.

Reply:
The idea of this plugin (and it's actions) was to provide a way to use object-management from process-links. We were aware that some actions already exist and are already part of various other plugins. During our refinement we discussed that those previous actions would be deprecated/removed in favour of this uniform way of working with objects.

If this is no longer the desired outcome, then please discuss this feature as a whole with Thomas Minke and let me know so I can stop working on this.

Reply:
@Klaas-Ritense we had an extensive discussion on this. We can't implement the actions in the objecten API plugin because we'd need a reference to the Object Management Configuration. That would be a circular dependency. That's why we're implementing a new plugin. @anna-ritense Please continue as discussed

Reply:
@ThomasMinkeRitense I agree with having a Object-Management-plugin. Just not for these plugins actions. crud actions belong in the objecten-api-plugin

A valid action for Object-Management-plugin would be:
`searchObjects("boom.leeftijd", ">=",  "12")`

Reply:
>A valid action for Object-Management-plugin would be:
searchObjects("boom.leeftijd", ">=",  "12")

There is a [story](https://ritense.tpondemand.com/entity/69068-be-object-management-plugin-action-retrieve) for this action, we also need a general get by Id for object-management configuration action. etc.